spec/support/slack_notifications.rb

```diff
@@ -0,0 +1,12 @@
+RSpec.configure do |config|
+  config.before do
+    stub_request(:post, "https://example.com/slack-webhook").
+      to_return(status: 200, body: "{}")
+  end
+
+  def expect_slack_message_with_text(text)
+    expect(WebMock).to have_requested(:post, "https://example.com/slack-webhook").with { |req|
```

Comment:
nit (and I know your views on `let` :)):

```
let(:slack_request) { stub_request(...) }
```

and then 

```
expect(slack_request).to_have_been_made.with( blah )
```

Reply:
Sorry!

```
       `let!` is not available from within an example (e.g. an `it` block) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc). It is only available on an example group (e.g. a `describe` or `context` block).
```

Reply:
I live to stand corrected